api/agent/agent.go

```diff
@@ -832,11 +833,21 @@ func (a *agent) prepCold(ctx context.Context, call *call, tok ResourceToken, ch
 
 	call.containerState.UpdateState(ctx, ContainerStateIdle, call.slots)
 
-	slot := &coldSlot{cookie, tok, err}
+	closer := func() {
+		if cookie != nil {
+			cookie.Close(ctx)
```

Comment:
in general this isn't great, we're enclosing a stale context from creation into `cookie.Close` when it's called much later, perhaps with a different enclosing context at the time

Reply:
not just for timeout, but for stats/trace ordering it gets really weird too...

Reply:
it seems like this is cold only after looking, so this is ok really if hot doesn't have this shortcoming

Reply:
yes, cold only and n/a for hot.